ports/stm32/ulpi.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2016 STMicroelectronics. All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ * USB3320 ULPI functions ported from stm32f7xx_lp_modes.c
```

Comment:
Can you specify which repo and directory this file was adapted from?

Reply:
https://github.com/STMicroelectronics/STM32CubeF7/blob/master/Projects/STM32746G-Discovery/Examples/PWR/PWR_CurrentConsumption/Src/stm32f7xx_lp_modes.c

Reply:
I think it's just one of the examples, I don't see any advanced examples ? And I actually took it from the older release (v1.15.0), which had that license in the file:

https://github.com/STMicroelectronics/STM32CubeF7/blob/v1.15.0/Projects/STM32746G-Discovery/Examples/PWR/PWR_CurrentConsumption/Src/stm32f7xx_lp_modes.c

That file, at that time was released under that license.

Reply:
OK, sounds good.

> Should I link that file in the ulpi.c ?

Yes please, put that exact link in.

Reply:
Should the license headers say ?

```c
/*
 * This file is part of the MicroPython project, http://micropython.org/
 * 
 * Copyright (c) 2016 STMicroelectronics. All rights reserved.
 *
 * This software component is licensed by ST under BSD 3-Clause license,
....
```

Reply:
Yes I think it makes sense to add the "This file is part of the MicroPython project, http://micropython.org/" line, because it was specifically customised for uPy.

Reply:
Done.